src/main/java/de/tum/in/www1/artemis/service/notifications/ConversationNotificationService.java

```diff
@@ -89,7 +95,13 @@ private void saveAndSend(ConversationNotification notification, Post createdMess
         singleUserNotificationRepository.saveAll(mentionedUserNotifications);
         mentionedUserNotifications.forEach(singleUserNotification -> websocketMessagingService.sendMessage(singleUserNotification.getTopic(), singleUserNotification));
 
-        sendNotificationViaWebSocket(notification, recipients.stream().filter(recipient -> !mentionedUsers.contains(recipient)).collect(Collectors.toSet()));
+        var conversationParticipants = conversationParticipantRepository.findConversationParticipantsByConversationIdAndUserIds(notification.getConversation().getId(),
+                recipients.stream().map(User::getId).collect(Collectors.toSet()));
+        var mutedUsers = conversationParticipants.stream().filter(ConversationParticipant::getIsMuted).map(ConversationParticipant::getUser).collect(Collectors.toSet());
+        var unmutedUsers = recipients.stream().filter(user -> !mutedUsers.contains(user)).collect(Collectors.toSet());
+        var unmutedAndUnmentionedUsers = unmutedUsers.stream().filter(user -> !mentionedUsers.contains(user)).collect(Collectors.toSet());
+
+        sendNotificationViaWebSocket(notification, unmutedAndUnmentionedUsers);
```

Comment:
Instead of performing an additional db query here, you can update the ConversationMessagingService.filterNotificationRecipients method and replace e.g. `summary.isConversationHidden()` with something like `summary.isConversationMuted()`.

Reply:
Thanks! bb10f27eabe11ef1430deab52fd4c23b7ee82b23